modules/plugin/coverage-multidim/netcdf/src/test/java/org/geotools/coverage/io/netcdf/NetCDFMosaicReaderTest.java

```diff
@@ -231,7 +231,7 @@ public void testHeterogeneous()
         File nc1 = TestData.file(this, "polyphemus_20130301_test.nc");
         File mosaic = new File(TestData.file(this, "."), "nc_poly_hetero");
         if (mosaic.exists()) {
-            FileUtils.deleteDirectory(mosaic);
+            FileUtils.forceDelete(mosaic);
```

Comment:
So all these changes from deleteDirectory to forceDelete, are they there to ensure an exception is thrown in case the deletion does not work? Or... something else?

Reply:
During the running of the tests I noticed they were sometimes struggling with the deleteDirectory. I'm not sure why tbh, I guess some other tests are influencing this one. I found it hard to locate the tests that were influencing this one, so I tried forceDeleting it. That solved the problem for me. It's a recurring theme in the tests that they sometimes don't cleanup properly.